### PR TITLE
Develop release workflow

### PR DESCRIPTION
**Pull request by byrnHDF**
This adds a SNAPSHOT tag and creates artifacts for download (location/url still under investigation).
Will also add doxygen artifacts before merging

### CODE REVIEW

File: .github/workflows/daily-build.yml [hunk 1]
```diff
@@ -6,9 +6,6 @@ on:
   schedule:
     - cron: "6 0 * * *"
 
-permissions:
-  contents: read
```
Comment (jhendersonHDF) [inline .github/workflows/daily-build.yml:10]: Does this action need write permissions? Can those be given on a per-job basis while leaving this line in?

Comment (byrnHDF) [reply .github/workflows/daily-build.yml:10]: Maybe - It might be from debugging. I will try

Comment (byrnHDF) [reply .github/workflows/daily-build.yml:10]: No - a deeper understanding of permissions will be needed.

Comment (jhendersonHDF) [reply .github/workflows/daily-build.yml:10]: I'm guessing that if you leave in the global read permissions line here, you still need to set write permissions for each job that needs it in this file, even if the workflows for those jobs are setting that.

**Review by derobins — APPROVED** [review]

**Review by jhendersonHDF — APPROVED** [review]

**Pull request merged**